File: src/main/java/com/lanka_guide/tripplanner/model/activity/transport/method/Flight.java
package com.lanka_guide.tripplanner.model.activity.transport.method;

import com.lanka_guide.tripplanner.model.place.transport.Airport;

import java.time.Duration;
import java.time.ZonedDateTime;

public class Flight extends TransportMethod {
    Airport departureAirport;
    Airport arrivalAirport;
    String code;
    ZonedDateTime flightDepartureTime;
    ZonedDateTime flightArrivalTime;
    ZonedDateTime airportArrivalTime;
    ZonedDateTime airportDepartureTime;
    Duration duration;

    public Flight(Airport departureAirport, Airport arrivalAirport, String code, ZonedDateTime from, ZonedDateTime to) {
        super(from, to);
        this.departureAirport = departureAirport;
        this.arrivalAirport = arrivalAirport;
        this.code = code;
    }
}

File: src/main/kotlin/com/lankaguide/tripplanner/model/activity/transport/flight/Flight.kt
package com.lankaguide.tripplanner.model.activity.transport.flight

import com.lankaguide.tripplanner.model.activity.transport.Transport
import com.lankaguide.tripplanner.model.place.transport.Airport

import java.time.Duration
import java.time.ZonedDateTime

data class Flight(val departureAirport: Airport, val arrivalAirport: Airport,
                  val code: String, val depature: ZonedDateTime,
                  val arrival: ZonedDateTime, val type: Type,
                  val connection: Connection? = null) :
        Transport(departureAirport, arrivalAirport, depature, arrival) {

    val duration: Duration = Duration.between(arrival, depature)

    enum class Type {
        DOMESTIC,
        INTERNATIONAL
    }

    class Connection(val to : Type?)
}